docs/LANGUAGE.md

```diff
@@ -0,0 +1,388 @@
+
+OPA includes a policy engine which is purpose built for reasoning about information represented in structured documents such as JSON. Data stored in in the policy engine can be queried using OPA's native query language: Opalog.
+
+## What is Opalog?
+
+Opalog is a modernized version of [Datalog](https://en.wikipedia.org/wiki/Datalog) which is a well understood, decades old query language. Opalog extends Datalog to support structured document models such as JSON. Opalog queries are assertions on data stored in OPA. These queries can be used to define policies which enumerate violations that exist in the data stored in OPA.
+
+## Why use Opalog?
+
+Use Opalog for defining policy which is easy to read and write. 
+
+Opalog focuses on providing powerful support for referencing nested documents and ensuring that queries are correct and unambiguous.
+
+Opalog is declarative so policy authors can focus on what queries should return rather than how queries should be executed. These queries are simpler and more concise than the equivalent in an imperative language. Like other applications which support declarative query languages, OPA is able to optimize queries to improve performance, e.g, indexing, concurrent evaluation, reordering, etc.
+
+## Opalog Grammar
+
+Opalog's syntax is defined by the following grammar:
+
+```
+module         ::= package { import } policy
+package        ::= "package" var { "." var }
+import         ::= "import" package [ "as" var ]
+policy         ::= { rule }
+rule           ::= rule-head ":-" rule-body
+rule-head      ::= rule-name [ rule-args ]
+rule-name      ::= var
+rule-args      ::= "[" [ key ] "]" [ "=" value ]
+key            ::= var
+value          ::= var
+rule-body      ::= [ literal { "," literal } ]
+literal        ::= expr | "not" expr
+expr           ::= term 
+                  | expr-builtin
+                  | expr-infix
+expr-builtin   ::= var "(" [ term { , term } ] ")"
+expr-infix     ::= term bool-operator term
+term           ::= ref | var | const
+bool-operator  ::= "=" | "!=" | "<" | ">" | ">=" | "<="
+ref            ::= ns { ref-arg }
+ns             ::= var
+ref-arg        ::= ref-arg-dot | ref-arg-brack
+ref-arg-brack  ::= "[" ( const | var ) "]"
+ref-arg-dot    ::= "." var
+var            ::= ALPHA { ALPHA | DIGIT | "_" }
+const          ::= scalar | object | array
+scalar         ::= STRING | NUMBER | TRUE | FALSE | NULL
+array          ::= "[" term { "," term } "]"
+object         ::= "{" object-item { "," object-item } "}"
+object-item    ::= term ":" term
+```
+
+The grammar defined above makes use of the following syntax:
+
+```
+[]     optional (zero or one instances)
+{}     repetition (zero or more instances)
+|      alteration (one of the instances)
+()     grouping (order of expansion)
+STRING JSON string
+NUMBER JSON number
+TRUE   JSON true 
+FALSE  JSON false
+NULL   JSON null
+ALPHA  ASCII characters A-Z and a-z
+DIGIT  ASCII characters 0-9
+```
+
+## Opalog Queries
```

Comment:
I'd be tempted to first describe how Opalog lets people query base documents (without differentiating base and virtual documents).  And then say that we can also use Opalog to construct new, virtual documents out of the documents we have been given.  It so happens that the query language you have been given suffices to define those new documents.